apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerConversionOptionsTests.cs

```diff
@@ -67,35 +67,41 @@ public void WithSpannerToClrMappings()
             Assert.Equal(SpannerDbType.Numeric, options.DecimalToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Date, options.DateTimeToConfiguredSpannerType);
 
-            options = options.WithClrToSpannerMappings("DecimalToPgNumeric,DateTimeToTimestamp");
+            options = options.WithClrToSpannerMappings("SingleToFloat64,DecimalToPgNumeric,DateTimeToTimestamp");
+            Assert.Equal(SpannerDbType.Float64, options.SingleToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.PgNumeric, options.DecimalToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Timestamp, options.DateTimeToConfiguredSpannerType);
 
             // Test with builder as well.
             var builder = new SpannerConnectionStringBuilder();
-            builder.ClrToSpannerTypeDefaultMappings = "DecimalToFloat64,DateTimeToDate";
+            builder.ClrToSpannerTypeDefaultMappings = "SingleToFloat32,DecimalToFloat64,DateTimeToDate";
+            Assert.Equal(SpannerDbType.Float32, builder.ConversionOptions.SingleToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Float64, builder.ConversionOptions.DecimalToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Date, builder.ConversionOptions.DateTimeToConfiguredSpannerType);
 
             // Explicitly set valid type mapping.
-            builder = new SpannerConnectionStringBuilder("ClrToSpannerTypeDefaultMappings=DecimalToNumeric,DateTimeToTimestamp");
+            builder = new SpannerConnectionStringBuilder("ClrToSpannerTypeDefaultMappings=SingleToFloat32,DecimalToNumeric,DateTimeToTimestamp");
+            Assert.Equal(SpannerDbType.Float32, builder.ConversionOptions.SingleToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Numeric, builder.ConversionOptions.DecimalToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Timestamp, builder.ConversionOptions.DateTimeToConfiguredSpannerType);
 
             // Check that properties are case-insensitive.
             builder = new SpannerConnectionStringBuilder();
-            builder.ClrToSpannerTypeDefaultMappings = "decimaltofloat64,datetimeTodate";
+            builder.ClrToSpannerTypeDefaultMappings = "singletofloat32,decimaltofloat64,datetimeTodate";
+            Assert.Equal(SpannerDbType.Float32, builder.ConversionOptions.SingleToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Float64, builder.ConversionOptions.DecimalToConfiguredSpannerType);
             Assert.Equal(SpannerDbType.Date, builder.ConversionOptions.DateTimeToConfiguredSpannerType);
         }
 
         [Theory]
         [InlineData("DecimalToFloat64,DecimalToNumeric,DateTimeToDate")] // Multiple mappings for a type.
         [InlineData("DecimalToFloat64,DecimalToPgNumeric,DateTimeToDate,DateTimeToTimestamp")] // Multiple mappings for a type.
+        [InlineData("SingleToFloat64,SingleToFloat32,DateTimeToDate,DateTimeToTimestamp")] // Multiple mappings for a type.
```

Comment:
I think all of these tests should be changed so that there's only *one* bad mapping involved... otherwise the fact that we've got DateTimeToDate and DateTimeToTimestamp together could be enough to cause the problem, even without us having the check for Single multiple mappings, if you see what I mean.

Reply:
Yep, you are right, done.